linera-client/src/chain_clients.rs

```diff
@@ -59,4 +59,8 @@ where
     pub async fn map_lock(&self) -> MutexGuard<ClientMapInner<P, S>> {
         self.0.lock().await
     }
+
+    pub async fn add_client(&self, client: ChainClient<P,S>) {
+        self.0.lock().await.insert(client.chain_id(), client);
+    }
```

Comment:
Based on the comment in `chain_listener.rs` this API cannot be used without some logic around it:
```
// For every entry in the client map we are already listening to notifications, so                                                                                                                                                           
// there's nothing to do. This can happen if we download a child before the parent                                                                                                                                                           
// chain, and then process the OpenChain message in the parent.
```

I think what you need here is a constructor to build a map with a single client.

Reply:
Not even that: Only the chain listener is supposed to add entries to the map, because it assumes that for each entry we've also already started listening to notifications.

But maybe we should change that behavior in the chain listener instead, now that we want to use it in other ways.

Reply:
I thnk I resolved the root cause here, so resolving for now.